pkg/tickers/retry.go

```diff
@@ -0,0 +1,94 @@
+package tickers
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+	"github.com/smartcontractkit/ocr2keepers/pkg/util"
+)
+
+const (
+	retryDelay              = 500 * time.Millisecond
+	totalAttempt            = 3
+	attemptsCacheExpiration = 10 * retryDelay // 5 seconds
+)
+
+type retryTicker struct {
+	timeTicker
+	nextRetries     sync.Map // time.Time -> ocr2keepers.UpkeepPayload
+	payloadAttempts *util.Cache[int]
+}
+
+// Retry adds a retryable result to the retryTicker.
+func (rt *retryTicker) Retry(result ocr2keepers.CheckResult) error {
+	payload := result.Payload
+
+	if result.Retryable {
+		attemptCount, ok := rt.payloadAttempts.Get(payload.ID)
+		if !ok {
+			attemptCount = 0
+		}
+
+		if attemptCount >= totalAttempt {
+			fmt.Printf("Payload %s has already been tried %d times. Skipping.\n", payload.ID, totalAttempt)
```

Comment:
Should this be using `log` instead of `fmt`? If so, we don't need the new line at the end.